src/RouterCLILookupCodegenBuilder.hack

```diff
@@ -72,11 +72,20 @@ final class RouterCLILookupCodegenBuilder {
       ->setShebangLine('#!/usr/bin/env hhvm')
       ->setFileType(CodegenFileType::HACK_PARTIAL)
       ->setGeneratedFrom($this->generatedFrom)
-      ->setPseudoMainHeader($this->getInitCode())
       ->addClass($this->getCodegenClass($router_classname, $utility_classname))
-      ->setPseudoMainFooterf(
-        '(new %s())->main($argv);',
-        $utility_classname,
+      ->addFunction(
+        $this->cg->codegenFunction('main')
```

Comment:
Given the uniqueness requirement, can you rename this to `hack_router_cli_lookup_generated_main()` or similar?